survey/forms.py

```diff
@@ -164,18 +163,15 @@ def get_question_initial(self, question, data):
         if answer:
             # Initialize the field with values from the database if any
             if question.type == Question.SELECT_MULTIPLE:
-                initial = []
                 if answer.body == "[]":
-                    pass
-                elif "[" in answer.body and "]" in answer.body:
                     initial = []
-                    unformated_choices = answer.body[1:-1].strip()
-                    for unformated_choice in unformated_choices.split(settings.CHOICES_SEPARATOR):
-                        choice = unformated_choice.split("'")[1]
-                        initial.append(slugify(choice))
```

Comment:
Here I was wondering why the contents of answer.body are not just JSON or some other standardized method. That way one doesn't run risks of this mechanism breaking as one can fall back to existing json deserialization methods.

It's not quite json as of now because it's using single quotation marks rather than double quotation marks. But it's already quite close. 

My code below doesn't change that but I think it's slightly more robust in some cases.

Reply:
This is a small sql database, so I suppose I wanted to avoid dumping a json.

Reply:
A standardized method would indeed be better.

Reply:
@Pierre-Sassoulas Ok, but it wouldn't take up any more space in the database. Instead of

```['one', 'two', 'three']```

in JSON that would simple be:

```["one", "two", "three"]```

An advantage would be that it would automatically be able to read things like this as well:

```["'one'", "t,w,o", "[three]"]```


I think my code below can do most of that as well though. So maybe it's not needed.

I spent a day last week cleaning up a ~90k row dump turning everything back from the slugified version to the original options because loaddata wouldn't accept the data before realizing that I needed to override this code as well to actually make the original data work. That's the reason I raise this.

Reply:
Sounds like it's going to be hard to make a migration from slugified to the new value (as there is less information in slugified answer than the actual answers) so putting that in a breaking change for version 2.0.0 and removing all the migrations files seems in order, what do you think ?

Reply:
Or maybe create a "best effort" migration file ?

Reply:
It should be possible to take all the options, convert them using slugify and that way make a pretty good guess as to which of the original options is the right one. Two cases where I don't think that's possible:

* If the options have changed after some respondents answered. The slugified version will either correspond to none of the current choices or may even correspond to an incorrect one.
* Cases where two or more options turn into the same slugified value.

So yeah, probably major version change and some way to gracefully handle the two above situations telling users that "sorry, some data was lost and we can no longer figure out what your respondents actually answered".

Reply:
Sounds good !